### PR TITLE
Callbacks and kernels

### PR DESCRIPTION
**Pull request by mateuszchudyk**

### CODE REVIEW

File: avx512_gemm.h [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #pragma once
 
+#include "config.h"
+
+#ifdef INTGEMM_COMPILER_SUPPORTS_AVX512
```
Comment (kpu) [inline avx512_gemm.h]: In general I like positive defines.  I'm a bit worried that somebody will include the code in another package, forget to define this, and not like the performance.  The nice thing about a negative define is it breaks the build if somebody screws up the defines.

File: vec_traits.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "types.h"
+
+namespace intgemm {
+
+/*
```
Comment (kpu) [inline vec_traits.h]: Maybe add a comment here that this is basically a lookup table?

**Comment by kpu** [comment]
We may need to rename the repo if there's a bunch of float and double kernels...

File: avx512_gemm.h (hunk 1 above)
Comment (mateuszchudyk) [reply avx512_gemm.h]: So maybe we should change this define and checking if it's true/false rather then defined/undefined and if it's undefined then just throw an error.

**Pull request merged**